metadeploy/api/views.py

```diff
@@ -309,6 +309,8 @@ def scratch_org_post(self, request, pk=None):
         uuid = str(uuid4())
         scratch_org = serializer.save(uuid=uuid)
         request.session["scratch_org_id"] = uuid
+        # @@@ TODO: This doesn't seem to persist immediately; race condition?
+        request.session.save()
```

Comment:
@wlonk None of the `self.scope["session"]` checks in `consumers.py` seem to be working for me, because (apparently?) the `request.session["scratch_org_id"] = uuid` in `views.py` has not taken effect in time? I tried adding this `request.session.save()`, but it doesn't seem to help. Basically when I create a new scratch org, I'm seeing `self.scope["session"].get("scratch_org_id")` in the consumer as an old uuid, not the newly-created one.

Reply:
Relevant docs:
 - https://docs.djangoproject.com/en/3.1/topics/http/sessions/#when-sessions-are-saved
 - https://channels.readthedocs.io/en/latest/topics/sessions.html#session-persistence

Looks to me like the session should be saved on line 311 anyway. Can you tell me about when the frontend talks with the consumer? As I understand it, there shouldn't be a race condition anyway, but I bet I'm misunderstanding what you're doing!

Reply:
Where are the transaction boundaries in the consumer? If it's reading the session in a transaction that started before an API call updated the session via a normal HTTP view, it makes sense that it wouldn't be able to see the updated session, due to MVCC.

Reply:
> Can you tell me about when the frontend talks with the consumer?

@wlonk After the initial `POST` returns successfully indicating that a new org is being spun up, the front end tries to subscribe to that `scratchorg` model. If you insert `print(uuid)` or `print(request.session["scratch_org_id"])` here in the view and then insert another `print(self.scope["session"].get("scratch_org_id", None))` at the beginning of `has_good_permissions` in `consumers.py`, you'll see that the consumer session does not have the updated uuid in the session.